### DISCUSSION TITLE
Documentation is vague

### DISCUSSION THREAD

**Discussion by Remonell** [selectize/selectize.js#1703] [Q&A]
I was messing around with selectize for the last couple of hours and I could not find any place where it is described, that selectize attaches to **already existing select/input DOMs**. All the demos and documentations made it seem to me, that the selectize function will create a select field with that id. For example: (taken from the [official website](https://selectize.dev/demos/2015/01/01/single-item-select/))
```
$("#select-beast").selectize({
  create: true,
  sortField: "text",
});
```
No mentions, that this element existed before.

Or in the[ usage documentation](https://github.com/selectize/selectize.js/blob/master/docs/usage.md):

```
<script type="text/javascript" src="selectize.js"></script>
<link rel="stylesheet" type="text/css" href="selectize.css" />
<script>
$(function() {
	$('select').selectize(options);
});
</script>
```
Same again.
I think this is misleading and will result in a lot of frustration for a lot of people. 

But maybe I am wrong? Is it necessary that a HTML select/input element exists beforehand?

**Comment by risadams** [comment]
In general, yes an element must exist before Selectize can initialize.
If an item needs to be created dynamically you can, but it should be inserted into the dom, and then selectize be called on the new element.

The documentation is definitely not the greatest, and updates are on the roadmap for when time permits.
